src/rules/no_external_imports.rs

```diff
@@ -0,0 +1,142 @@
+  Unexpected,
+}
+
+#[derive(Display)]
+enum NoExternalImportHint {
+  #[display(fmt = "Create a deps.ts file or use import maps")]
+  CreateDependencyFile,
+}
+
+impl LintRule for NoExternalImport {
+  fn new() -> Arc<Self> {
+    Arc::new(NoExternalImport)
+  }
+
+  fn tags(&self) -> &'static [&'static str] {
+    &[]
+  }
+
+  fn code(&self) -> &'static str {
+    CODE
+  }
+
+  fn lint_program(&self, _context: &mut Context, _program: ProgramRef) {
+    unreachable!();
+  }
+
+  fn lint_program_with_ast_view(
+    &self,
+    context: &mut Context,
+    program: Program,
+  ) {
+    let mut handler = NoExternalImportHandler::default();
+    handler.traverse(program, context);
+  }
+
+  #[cfg(feature = "docs")]
+  fn docs(&self) -> &'static str {
+    include_str!("../../docs/rules/no_external_imports.md")
+  }
+}
+
+#[derive(Default)]
+struct NoExternalImportHandler;
+
+impl NoExternalImportHandler {
+  fn check_import_path<'a>(&'a self, decl: &ImportDecl, ctx: &mut Context) {
+    let parsed_src = Url::parse(decl.src.value());
+    let file_name = Path::new(ctx.file_name())
+      .file_stem()
+      .and_then(OsStr::to_str);
+
+    if parsed_src.is_ok() && file_name != Some("deps") {
+      ctx.add_diagnostic_with_hint(
+        decl.span(),
+        CODE,
+        NoExternalImportMessage::Unexpected,
+        NoExternalImportHint::CreateDependencyFile,
+      );
+    }
+  }
+}
+
+impl Handler for NoExternalImportHandler {
+  fn import_decl(
+    &mut self,
+    decl: &deno_ast::view::ImportDecl,
+    ctx: &mut Context,
+  ) {
+    self.check_import_path(decl, ctx);
+  }
+}
+
+#[cfg(test)]
+mod tests {
+  use super::*;
+
+  #[test]
+  fn no_external_import_valid() {
+    assert_lint_ok! {
+      NoExternalImport,
+      "import { assertEquals } from './deps.ts'",
+      "import { assertEquals } from 'deps.ts'",
+      "import Foo from './deps.ts';",
+      "import type { Foo } from './deps.ts';",
+      "import type Foo from './deps.ts';",
+      "import * as Foo from './deps.ts';",
+      "import './deps.ts';",
+      "const foo = await import('https://example.com');"
+    };
+
+    assert_lint_ok! {
+      NoExternalImport,
+      filename: "deps.ts",
+      "import { assertEquals } from 'https://deno.land/std@0.126.0/testing/asserts.ts'"
+    };
+  }
+
+  #[test]
+  fn no_external_import_invalid() {
+    assert_lint_err! {
+      NoExternalImport,
+      "import { assertEquals } from 'https://deno.land/std@0.126.0/testing/asserts.ts'": [
+        {
+          col: 0,
+          message: NoExternalImportMessage::Unexpected,
+          hint: NoExternalImportHint::CreateDependencyFile,
+        },
+      ],
+    };
+
+    assert_lint_err! {
+      NoExternalImport,
+      "import assertEquals from 'http://deno.land/std@0.126.0/testing/asserts.ts'": [
+        {
+          col: 0,
+          message: NoExternalImportMessage::Unexpected,
+          hint: NoExternalImportHint::CreateDependencyFile,
+        },
+      ],
+    };
```

Comment:
Could you merge these tests into one `assert_lint_err!` so we can make sure this follows the same code style as other tests?
Also, it would be super nice to have more tests here - I'm worried that the shortage of invalid test cases would lead to false negatives.
I think the following 4 cases, which are brought from valid cases and their import paths are replaced with some URL, would be enough.

```ts
"import type { Foo } from 'https://example.com';",
"import type Foo from 'https://example.com';",
"import * as Foo from 'https://example.com';",
"import 'https://example.com';",
```